lib/base-service.js

```diff
@@ -0,0 +1,244 @@
+            CrudBaseService.throwNotFoundError();
+          }
+          const fieldName = BaseService.getSubDocumentFieldName(call.request);
+          model[fieldName].forEach((child) => {
+            model[fieldName].id(child._id).remove();
+          });
+          call.request[fieldName].forEach((child) => {
+            model[fieldName].push(child);
+          });
+          return model.save();
+        }).then(() => {
+          return {};
+        });
+    },
+  };
+}
+
+function getRelatedModelMethods(Model) {
+  return {
+    'push': function(call) {
+      return Model.findOne({'_id': call.request.id})
+        .then((model) => {
+          if (model === null) {
+            CrudBaseService.throwNotFoundError();
+          }
+          const fieldName = BaseService.getSubDocumentFieldName(call.request);
+          call.request[fieldName].forEach((relatedModel) => {
+            model[fieldName].push(relatedModel.id);
+          });
+          return model.save();
+        }).then(() => {
+          return {};
+        });
+    },
+    'addToSet': function(call) {
+      return Model.findOne({'_id': call.request.id})
+        .then((model) => {
+          if (model === null) {
+            CrudBaseService.throwNotFoundError();
+          }
+          const fieldName = BaseService.getSubDocumentFieldName(call.request);
+          call.request[fieldName].forEach((relatedModel) => {
+            model[fieldName].addToSet(relatedModel.id);
+          });
+          return model.save();
+        }).then(() => {
+          return {};
+        });
+    },
+    'remove': function(call) {
+      return Model.findOne({'_id': call.request.id})
+        .then((model) => {
+          if (model === null) {
+            CrudBaseService.throwNotFoundError();
+          }
+          const fieldName = BaseService.getSubDocumentFieldName(call.request);
+          call.request[fieldName].forEach((relatedModel) => {
+            if (!existRelatedModel(model[fieldName], relatedModel)) {
+              /* eslint-disable no-console */
+              console.warn(`related model id '${relatedModel.id}' does not exist in 
+              '${fieldName}'`);
+              return;
+            }
+            model[fieldName].remove(relatedModel.id);
+          });
+          return model.save();
+        }).then(() => {
+          return {};
+        });
+    },
+    'replace': function(call) {
+      return Model.findOne({'_id': call.request.id})
+        .then((model) => {
+          if (model === null) {
+            CrudBaseService.throwNotFoundError();
+          }
+          const fieldName = BaseService.getSubDocumentFieldName(call.request);
+          model[fieldName] = [];
+          call.request[fieldName].forEach((relatedModel) => {
+            model[fieldName].push(relatedModel.id);
+          });
+          return model.save();
+        }).then(() => {
+          return {};
+        });
+    },
+  };
+}
+
+function existChild(children, child) {
```

Comment:
Se puede refactorizar y prescindir de la variable existChild, se puede cambiar a hacer un for normal para que pueda hacer return true en vez de recorrer todo el arreglo incluso si ya encontró el id, otra manera sería filtrar por la condición del if y retornar dependiendo del .length del arreglo resultante